scripts/build_baremetal.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/sh
+
+# all.sh
+#
+# This file is part of mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2014-2016, ARM Limited, All Rights Reserved
+#
+# Purpose
+#
+# To build for bare metal with given toolchain
+#
+# Warning: the test is destructive. It includes various build modes and
+# configurations, and can and will arbitrarily change the current CMake
+# configuration. After this script has been run, the CMake cache will be lost
+# and CMake will no longer be initialised.
+#
+
+#  Parse args for compiler and check they are in path
+
+CONFIG_H='include/mbedtls/config.h'
+CONFIG_BAK="$CONFIG_H.bak"
+
+usage()
+{
+    printf "Usage: $0\n"
+    printf "  -h|--help\t\tPrint this help.\n"
+    printf "  -c|--cc\t\tSupply compiler command\n"
+    printf "  -a|--ar\t\tSupply archive Command\n"
+}
+
+err_msg()
+{
+    echo "$1" >&2
+}
+
+check_tools()
+{
+    for TOOL in "$@"; do
+        if ! `which "$TOOL" >/dev/null 2>&1`; then
+            err_msg "$TOOL not found!"
+            exit 1
+        fi
+    done
+}
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+        --cc|-c*)
+            shift
+            CC="$1"
+            ;;
+        --ar|-a*)
+            shift
+            AR="$1"
+            ;;
+        --help|-h|*)
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+cp "$CONFIG_H" "$CONFIG_BAK"
```

Comment:
> I found it useful to leave the modified file. specially for on-target testing.

I dont know what are the requirements of the script, so this might be the right thing to do. Normally, we do clean up and put everything as it was at the end of the script. You can check the other test programs.

> Also if we are putting -eu to exit on error. Than reverting to backup may not work.

You can use a trap, just like in all.sh https://github.com/ARMmbed/mbedtls/blob/development/tests/scripts/all.sh#L90